doc/migrations/v0.43-v0.44.md

```diff
@@ -0,0 +1,117 @@
+# Migrating to libp2p@44 <!-- omit in toc -->
+
+A migration guide for refactoring your application code from libp2p v0.43.x to v0.44.0.
+
+## Table of Contents <!-- omit in toc -->
+
+- [Connection Manager](#connection-manager)
+- [Connection Gater](#connection-gater)
+
+## Connection Manager
+
+The connection manager has been refactored to be simpler and faster.  The changes
+are mostly internal but some configuration options have changed.
+
+**Before**
+
+```js
+import { createLibp2p } from 'libp2p'
+
+const node = await createLibp2p({
+  connectionManager: {
+    // this option has been removed as auto dial now runs in response to peers
+    // disconnecting and is no longer on a timer
+    autoDialInterval: 0,
+
+    // this has been renamed to `maxPeerAddrsToDial`
+    maxAddrsToDial: 20,
+
+    // this has been renamed to `maxParallelDialsPerPeer`
+    maxDialsPerPeer: 20,
+
+    // these should now be ranges expressed as MultiaddrFilters
+    allow: [
+      '/ip4/0.0.0.0/tcp/123'
+    ],
+
+    // these should now be ranges expressed as MultiaddrFilters
+    deny: [
+      '/ip4/0.0.0.0/tcp/123'
+    ]
+  }
+})
+```
+
+**Before**
+
+```js
+import { createLibp2p } from 'libp2p'
+import { MultiaddrFilter } from from '@multiformats/multiaddr'
+
+const node = await createLibp2p({
+  connectionManager: {
+    // how many peers to dial at once while trying to ensure the node
+    // is above minConnections
+    autoDialConcurrency: 25,
+
+    // a low value allows user-initiated dials to take priority over
+    // auto dials
+    autoDialPriority: 0,
+
+    // this was previously named `maxAddrsToDial`
+    maxPeerAddrsToDial: 20,
+
+    // this was previously named `maxDialsPerPeer`
+    maxParallelDialsPerPeer: 20,
+
+    // these should now be ranges expressed as MultiaddrFilters
+    allow: [
+      new MultiaddrFilter('/ip4/192.168.1.1/ipcidr/32')
+    ],
+
```

Comment:
As above, can i create a multiaddrFilter consisting only of '/p2p/peerId' ?

Reply:
No, the allow/deny lists are for allowing/denying classes of network to make connections - this happens before the peer id exchange so we don't know who the remote peer is at that point.

It's a DoS mitigation strategy to always allow connections from networks you know, control or trust, or disallow connections from networks that are attacking you.

To deny specific peers you can implement a [connection gater](https://github.com/libp2p/js-libp2p/blob/master/doc/CONFIGURATION.md#configuring-connection-gater).